api/include/opentelemetry/nostd/string_view.h

```diff
@@ -1,13 +1,30 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 #pragma once
 
-#include <algorithm>
-#include <cstddef>
-#include <cstring>
-#include <ostream>
-#include <stdexcept>
-#include <string>
+#ifdef HAVE_CPP_STDLIB
+#  include "opentelemetry/std/string_view.h"
+#else
```

Comment:
Check for HAVE_ABSEIL ?

Reply:
Not yet. I'm not adding full `HAVE_ABSEIL` support, planning to only use that for `absl::variant` with Visual Studio 2015. 

In all other build systems we prefer the following - in the following order:
- standard classes in C++17
- `gsl::span` for C++11, C++14, C++17 (as GSL is the only fully conforming implementation)
- `absl::variant` for C++11 (vs2015, because Linux and Mac are fine without it already with MPark variant).. and for C++14. Then C++17 already has `std::variant`.